ee/clickhouse/queries/funnels/test/breakdown_cases.py

```diff
@@ -1366,4 +1369,347 @@ def test_basic_funnel_default_funnel_days_breakdown_action(self):
+                event="sign up",
+                distinct_id="person3",
+                properties={"$group_0": "org:6", "$browser": "Safari"},
+                timestamp="2020-01-02T14:00:00Z",
+            )
+
+            result = funnel.run()
+            self.assertEqual(
+                result[0],
+                [
+                    {
+                        "action_id": "sign up",
+                        "name": "sign up",
+                        "custom_name": None,
+                        "order": 0,
+                        "people": [person1.uuid] if Funnel == ClickhouseFunnel else [],  # backwards compatibility
+                        "count": 1,
+                        "type": "events",
+                        "average_conversion_time": None,
+                        "median_conversion_time": None,
+                        "breakdown": "finance",
+                        "breakdown_value": "finance",
+                    },
+                    {
+                        "action_id": "play movie",
+                        "name": "play movie",
+                        "custom_name": None,
+                        "order": 1,
+                        "people": [person1.uuid] if Funnel == ClickhouseFunnel else [],  # backwards compatibility
+                        "count": 1,
+                        "type": "events",
+                        "average_conversion_time": 3600.0,
+                        "median_conversion_time": 3600.0,
+                        "breakdown": "finance",
+                        "breakdown_value": "finance",
+                    },
+                    {
+                        "action_id": "buy",
+                        "name": "buy",
+                        "custom_name": None,
+                        "order": 2,
+                        "people": [person1.uuid] if Funnel == ClickhouseFunnel else [],  # backwards compatibility
+                        "count": 1,
+                        "type": "events",
+                        "average_conversion_time": 7200.0,
+                        "median_conversion_time": 7200.0,
+                        "breakdown": "finance",
+                        "breakdown_value": "finance",
+                    },
+                ],
+            )
+            # Querying persons when aggregating by persons should be ok, despite group breakdown
+            self.assertCountEqual(self._get_people_at_step(filter, 1, "finance"), [person1.uuid])
+            self.assertCountEqual(self._get_people_at_step(filter, 2, "finance"), [person1.uuid])
+            self.assertEqual(
+                result[1],
+                [
+                    {
+                        "action_id": "sign up",
+                        "name": "sign up",
+                        "custom_name": None,
+                        "order": 0,
+                        "people": [person2.uuid, person3.uuid]
+                        if Funnel == ClickhouseFunnel
```

Comment:
Q: Why do we have if/else clauses in these tests?

Reply:
these cases run for all types of funnels: strict, unordered, and steps funnel. Since steps funnel existed beforehand, we had backwards compatible API changes that returned people as an array in the results.

For the newer ones, we didn't keep this contract in place. This seemed like the smallest cleanest way to incorporate things in for common tests